ssh-pairing-server.c

```diff
@@ -37,6 +66,16 @@ int main(int argc, char *argv[])
 
 	// Client identifier
 	char clientname[INET6_ADDRSTRLEN];
+	/*
+	 * what is the purpose of this "ssh-pairing" default value of
```

Comment:
It's for the comment part of the authorized_keys line. Normally that's the hostname of the client system, but I wanted to avoid the potential issues of a reverse DNS lookup

Reply:
Obtaining the numerical IP address of the peer should practically never fail on a connected socket, aside from out of memory situations and alike. It might be better to exit with an error code if no numerical IP address can be obtained. Then no default name is required.

It could be worth a thought to add a command line option to pass the desired hostname for the authorized_keys file. This would make the file more descriptive than numerical IP addresses or a default name.

**If** `clientname` is also used to store hostnames then it might make more sense to use the `NI_MAXHOST` constant to dimension the `clientname` string. At least this would put less of a questionmark in my mind, since placing a numerical IP address within `NI_MAXHOST` bytes will always be possible, while placing a hostname in `INET6_ADDRSTRLEN` bytes won't.

Reply:
> Obtaining the numerical IP address of the peer should practically never fail on a connected socket, aside from out of memory situations and alike. It might be better to exit with an error code if no numerical IP address can be obtained. Then no default name is required.

Can be done, but that terminates the connection which is a bit annoying. It seemed easier for me to just add a fallback.

> It could be worth a thought to add a command line option to pass the desired hostname for the authorized_keys file. This would make the file more descriptive than numerical IP addresses or a default name.

Could be done, but IMO the caller could simply replace the whole "comment" column of the output in that case. Initially I targeted ssh-pairing for the initial enrollment for a server, which should require as little input as possible, but it might also be a good usecase for subsequent enrollments from other clients later.

> If clientname is also used to store hostnames then it might make more sense to use the NI_MAXHOST constant to dimension the clientname string. At least this would put less of a questionmark in my mind, since placing a numerical IP address within NI_MAXHOST bytes will always be possible, while placing a hostname in INET6_ADDRSTRLEN bytes won't.

Ok, will do.